File: model/Room.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ETT.model
{
    public class Room
    {
        public enum RoomType
        {
            SMALL,
            MEDIUM,
            LARGE
        }
        public RoomType fromString(string param)
        {
            string[] values = Enum.GetNames(typeof(RoomType));
            foreach (string s in values)
            {
                if (s == param)
                {
                    Console.WriteLine(s);
                }
            }
            return 0;
        }
        private string room { get; set; }
        private RoomType type { get; set; }

        public Room() {
        }

        public Room(string room, RoomType type)
        {
            this.room = room;
            this.type = type;
        }

        // @Enumerated(EnumType.STRING)
        public String getRoom() {
            return room;
        }
        public void setRoom(String room) {
            this.room = room;
        }
        public RoomType getType() {
            return type;
        }
        public void setType(RoomType type) {
            this.type = type;
        }
    }
}

File: ExaminationTimetabling/src/main/java/com/timetabling/app/model/Room.java
package com.timetabling.app.model;

import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnore;

import com.timetabling.app.enums.RoomType;

public class Room {

	 @JsonProperty("Room")
	 private String Room;
	 @JsonProperty("Type")
	 private RoomType type;
	 
	 public Room() {}
	 
	 @JsonIgnore
	 @Enumerated(EnumType.STRING)
	 public String getRoom() {
		 return Room;
	 }

	 @JsonIgnore
	 public void setRoom(String Room) {
		 this.Room = Room;
	 }

	 @JsonIgnore
	 public RoomType getType() {
		 return type;
	 }

	 @JsonIgnore
	 public void setType(RoomType type) {
		 this.type = type;
	 }
	 
	 public Room(Builder builder) {
		 this.Room = builder.Room;
		 this.type = builder.type;
	 }
	 
	 public static class Builder {
		 private String Room;
		 private RoomType type;
		 
		 public Builder number(String Room) {
			 this.Room = Room;
			 return this;
		 }
		 
		 public Builder type(RoomType type) {
			 this.type = type;
			 return this;
		 }
		 
		 public Room build() {
			 return new Room(this);
		 }
	 }
}	
